master/buildbot/steps/source/git.py

```diff
@@ -362,14 +362,14 @@ def _clone(self, shallowClone):
         """Retry if clone failed"""
 
         command = ['clone']
-        switchToBranch = False
+        switchToBranch = True
```

Comment:
Shouldn't be `True`, when `self.supportsBranch` is `False`. Wouldn't it be right to set it to `self.supportsBranch`?

Reply:
`switchToBranch` must be `True` when `self.supportsBranch` is `False`.
Otherwise no post-clone branch switching happens when `supoortsBranch` is `False`; which is one of the issues I was having.

Reply:
... though `switchToBranch = True` is redundant for 'HEAD' which should be fixed now